zio-cache/shared/src/test/scala/zio/cache/ScopedCacheSpec.scala

```diff
@@ -721,8 +753,137 @@ object ScopedCacheSpec extends ZIOSpecDefault {
+                            _                   <- clock.advance(500.milliseconds)
+                            freeAfter500ms      <- cache.freeExpired
+                            cacheSizeAfter500ms <- cache.size
+                            onlyCorrectResourceCleanedAt500ms <-
+                              observablesResource(0).assertAcquiredOnceAndCleaned && observablesResource(
+                                1
+                              ).assertAcquiredOnceAndNotCleaned && observablesResource(
+                                2
+                              ).assertAcquiredOnceAndNotCleaned
+                            _               <- clock.advance(1.second)
+                            freeAfter1500ms <- cache.freeExpired
+                            onlyCorrectResourceCleanedAt1500ms <-
+                              observablesResource(1).assertAcquiredOnceAndCleaned && observablesResource(
+                                2
+                              ).assertAcquiredOnceAndNotCleaned
+                            cacheSizeAfter1500ms        <- cache.size
+                            _                           <- clock.advance(1.second)
+                            freeAfter2500ms             <- cache.freeExpired
+                            cacheSizeAfter2500ms        <- cache.size
+                            allResourcesCleanedAt2500ms <- observablesResource(2).assertAcquiredOnceAndCleaned
+                          } yield assertTrue(
+                            freeAfter500ms == 1 && cacheSizeAfter500ms == 2 &&
+                              freeAfter1500ms == 1 && cacheSizeAfter1500ms == 1 &&
+                              freeAfter2500ms == 1 && cacheSizeAfter2500ms == 0
+                          ) && onlyCorrectResourceCleanedAt500ms && onlyCorrectResourceCleanedAt1500ms && allResourcesCleanedAt2500ms
+                        }
+                      }
+          } yield result
+        }
       )
     ),
+    suite("automatic clean expired resource")(
+      test("should once in a while clean and remove from the cache expired resource") {
+        val numResource = 30
+        for {
+          observablesResource <-
+            ZIO.foreach((0 until numResource).toList)(_ => ObservableResourceForTest.makeUnit)
+          clock <- MockedJavaClock.make
+          scopedCache =
+            ScopedCache.makeWith(
+              capacity = numResource,
+              scopedLookup = ScopedLookup((key: Int) => observablesResource(key).managed),
+              clock,
+              cleaningInterval = 10.second
+            )(timeToLive = { _: Exit[Nothing, Unit] => Duration.fromSeconds(40) })
+          result <- ZIO.scoped {
+                      scopedCache.flatMap { cache =>
+                        for {
+                          _ <- ZIO.foreachParDiscard((0 until numResource))(n => ZIO.scoped(cache.get(n).unit))
+                          _ <- clock.advance(30.second)
```

Comment:
Going to do a PR to make ScopedCache use ZIO clock everywhere

Reply:
That's fine. You can use the unsafe API now so you don't need to execute ZIO workflows to use it.

Reply:
Not sure to understand, the coded is fine like this? Or I need to modify something somewhere? In fact clock.advance is using the unsafe API.

Reply:
It is fine in the test here but if you are using the clock in the actual cache implementation you should use the 1`unsafe` operators that do not return a ZIO workflow for performance.

Reply:
I thought ZIO.clock in ZIO2 was performant enough. I use `.sleep(1.second)` on a effect, I am not sure how to do that in performant way with unsafe

Reply:
That's fine because it is asynchronous. But if you are doing a synchronous operation like getting the time you should use the unsafe variant so it is equivalent to just accessing the system time directly.

Reply:
Oh yes, I understand now!

Reply:
But in the long term, you don't think a PR making this cache use ZIO clock would be good?

Reply:
I think with ZIO 2 it is fine to use the ZIO clock internally in the cache implementation as long as we make sure to use the unsafe operators for synchronous operations like getting the time. This will improve testability and have no performance impact so it will be a net positive.